### PR TITLE
Fix TestsManifestGeneration tests

### LINKED ISSUE #46518: Test_EventSource_EtwManifestGenerationRollover failed in CI
net6.0-windows-Debug-x64-CoreCLR_release-Windows.81.Amd64.Open
https://dev.azure.com/dnceng/public/_build/results?buildId=936179&view=ms.vss-test-web.build-test-results-tab&runId=29613832&resultId=177187&paneView=debug
```
BasicEventSourceTests.TestsManifestGeneration.Test_EventSource_EtwManifestGenerationRollover

Error message
Microsoft.DotNet.RemoteExecutor.RemoteExecutionException : Remote process failed with an unhandled exception.

Child exception:
  Xunit.Sdk.TrueException: Assert.True() Failure
Expected: True
Actual:   False
   at BasicEventSourceTests.TestsManifestGeneration.<Test_EventSource_EtwManifestGenerationRollover>b__6_0() in /_/src/libraries/System.Diagnostics.Tracing/tests/BasicEventSourceTest/TestsManifestGeneration.Etw.cs:line 100
```

This particular failure also happened on https://github.com/dotnet/runtime/pull/46238 on windows.10.amd64.open.rt 

@sywhang looks like this is your test?

### PR DESCRIPTION
**Pull request by sywhang**
Fix #46518 

These tests sometimes fail because the call to disable the provider or roll over the file happens almost instantly after TraceEventSession.Flush(), and the Manifest data may not have been fully written to the file before we check for its existence. This adds a sleep between flushing and disabling the provider so that there's plenty of leeway for the payload to be fully written to the trace file.

### CODE REVIEW

File: src/libraries/System.Diagnostics.Tracing/tests/BasicEventSourceTest/TestsManifestGeneration.Etw.cs [hunk 1]
```diff
@@ -54,6 +54,9 @@ public void Test_EventSource_EtwManifestGeneration()
                     Thread.Sleep(TimeSpan.FromSeconds(5));
 
                     tracesession.Flush();
```
Comment (josalem) [inline src/libraries/System.Diagnostics.Tracing/tests/BasicEventSourceTest/TestsManifestGeneration.Etw.cs:56]: Is `Flush` asynchronous?

Comment (sywhang) [reply src/libraries/System.Diagnostics.Tracing/tests/BasicEventSourceTest/TestsManifestGeneration.Etw.cs:56]: The API itself isn't. The OS signal being received on the receiver end to flush the session and fully write the file is.

Comment (josalem) [reply src/libraries/System.Diagnostics.Tracing/tests/BasicEventSourceTest/TestsManifestGeneration.Etw.cs:56]: So calling `Flush` just tells ETW to flush its buffers, and _that_ is asynchronous from the `Flush` call.  Got it.

**Review by josalem — APPROVED** [review]
I don't like adding sleeps in tests, but that seems to be the pattern for this test already and as you've pointed out, we need to wait here anyway.  Can you decorate the sleep calls you're adding with comments for why they are "necessary"?

**Comment by danmosemsft** [comment]
Coming to this late, but could you not use the pattern of using a nominal (or no) sleep, doing the test, and if it fails, doing the longer sleep and testing again? This is the pattern we use many other places. Per the test history, we know that in the 99% case this sleep is not necessary. With the change as is, the tests all take longer to run.

**Comment by sywhang** [comment]
@danmosemsft thanks for the suggestion. So these sleeps were removed in a subsequent PR that addressed more issues on the test... But yes, I agree and I think we need to come up with a better synchronization mechanisms for the tracing tests as many of them are timing sensitive.

**Pull request merged**